Exercise-master/src/main/java/org/upgrad/controllers/UserController.java

```diff
@@ -0,0 +1,133 @@
+ */
+@RestController
+public class UserController {
+
+    @Autowired
+    private UserService userService ;
+
+    /*
+        This returns true if username already exists.
+        Otherwise it returns false.
+     */
+    public Boolean registerUserWithExistingUsername(String userName) throws Exception{
+        String userPresent = String.valueOf(userService.findUserByUsername(userName));
+        if ( ! (userPresent.equalsIgnoreCase("null") )) {
+            return true ;
+        } else  {
+            return false ;
+        }
+    }
+
+    /*
+        This returns true if email already exists.
+        Otherwise it returns false.
+     */
+    public Boolean registerUserWithExistingEmail(String email) throws Exception{
+        String userPresent = String.valueOf(userService.findUserByEmail(email));
+        if ( ! (userPresent.equalsIgnoreCase("null") )) {
+            return true ;
+        } else  {
+            return false ;
+        }
+    }
+
+
+    /*
+     It checks signup the user along with it's details if user is not already present.
+     This method gives relevant messages in case username or email is already registered.
+     */
+
+    @PostMapping("/api/user/signup/")
+    public ResponseEntity registerUser(@RequestParam String userName, @RequestParam String email, @RequestParam String password , @RequestParam String firstName, @RequestParam String lastName, @RequestParam String aboutMe, @RequestParam String contactNumber, @RequestParam String dob, @RequestParam String country) throws Exception{
+
+        // Setting url data into user model class.
+        User user = new User();
+        user.setPassword(hashPassword(password));
+        user.setLastName(lastName);
+        user.setFirstName(firstName);
+        user.setEmail(email);
+        user.setCountry(country);
+        user.setContactNumber(contactNumber);
+        user.setUserName(userName);
+        user.setDob(new SimpleDateFormat("dd/MM/yyyy").parse(dob));
+        user.setAboutMe(aboutMe);
+
+        String message = null ;
+
+       if ( registerUserWithExistingUsername(user.getUserName()) )
+       {
+            message = "Try any other Username, this Username has already been taken." ;
+            return new ResponseEntity<>( message , HttpStatus.OK) ;
+       }  else if  ( registerUserWithExistingEmail(user.getEmail())) {
+           message = "This user has already been registered, try with any other emailId." ;
+           return new ResponseEntity<>( message , HttpStatus.OK) ;
+       }  else {
+
+             //TODO : Make Parameters optional
+              userService.registerUserDetails(user);
+              message = userName + " successfully registered" ;
+              return new ResponseEntity<>( message , HttpStatus.OK) ;
+       }
+    }
+
+
+    @PostMapping("/api/user/login")
+    public ResponseEntity loginWithBadCredentials(@RequestParam String userName, @RequestParam String password)
```

Comment:
Can we change the name of the method since this method name should not impact tests.It can be something simple like login or loginUser

Reply:
It is in testcase